### ISSUE TITLE
Style nodes and arrows depending on type

### ISSUE THREAD

**Issue by brendanheywood** [catalyst/moodle-tool_dataflows#107]
All styles should be both a shape and not just a color, but also usually a color

All these styles I've made up on the spot (mostly looking at other engines and charts in the wild) the specifics aren't critical and we can evolve it later. Some of the combinations might be too cluttered so just implement and see and tweak.

* [x] Triggers should be oval / round pill boxes, and blue
* [x] Connectors should be square boxes and grey
* [x] Flow steps should be green and with small rounded corners
* [x] Connectors or flow steps which mutate state in the outside world are parallelograms instead of rectangle
* [x] Arrows connecting connectors should be a solid black arrow
* [x] Arrows Connecting flows should be a dotted arrow
* [ ] Nodes which 
* [x] Nodes which are invalid are shown in red

I think this should be largely automatic and handled by the manager code, but possibly we will need to have a method where a step can add some styles in graphvis format (want to avoid in case we change renderer for the graph)

**Comment by keevan** [comment]
There's one remaining point which might have been unintentional. This issue will close after #152 merges so please feel free to open another issue if you recall the missing item.
